### ISSUE TITLE
missing-param-doc for numpy doc style: variable with list of legal values

### ISSUE THREAD

**Issue by juliangilbey** [PyCQA/pylint#4035]
According to the [NumPy documentation style guidelines](https://numpydoc.readthedocs.io/en/latest/format.html), section 4:
> When a parameter can only assume one of a fixed set of values, those values can be listed in braces, with the default appearing first:
```
order : {'C', 'F', 'A'}
    Description of `order`.
```
However, using this syntax in a function docstring causes pylint to complain about `missing-param-doc` and `missing-type-doc`.

Here's a minimal example.  (With only one parameter of this form, it is completely ignored.)

```python
def myfunc(arg, option):
    """Do something boring.

    Parameters
    ----------
    arg : int
        The number of times to print it.
    option : {"y", "n"}
        Do I do it?

    """
    for i in range(arg):
        print(option)
```
and here is the `pylint` output:
```
************* Module pylint_test
pylint_test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
pylint_test.py:1:0: W9015: "option" missing in parameter documentation (missing-param-doc)
pylint_test.py:1:0: W9016: "option" missing in parameter type documentation (missing-type-doc)
pylint_test.py:12:8: W0612: Unused variable 'i' (unused-variable)

------------------------------------
Your code has been rated at -3.33/10
```

### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.9.1+ (default, Jan 10 2021, 15:42:50) 
[GCC 10.2.1 20201224]
```

**Comment by hippo91** [comment]
@juliangilbey thanks for this report.

**Comment by DanielNoord** [comment]
I would like to fix this after merging #5231
